solarforecastarbiter/metrics/preprocessing.py

```diff
@@ -213,47 +223,66 @@ def _resample_obs(obs, fx, obs_data, quality_flags):
     obs_flags['ISNAN'] = obs_data['value'].isna()
 
     # determine the points that should never contribute
-    discard_before_resample_flag_types = [
-        f.type for f in quality_flags if f.discard_before_resample
-    ]
-    discard_before_resample = obs_flags[discard_before_resample_flag_types]
+    # combine unique elements of tuple of tuples
+    discard_before_resample_flags = set(['ISNAN'])
+    for f in filter(lambda x: x.discard_before_resample, quality_flags):
+        discard_before_resample_flags |= set(f.quality_flags)
+    discard_before_resample = obs_flags[discard_before_resample_flags]
     counts = discard_before_resample.astype(int).sum(axis=0).to_dict()
     to_discard_before_resample = discard_before_resample.any(axis=1)
 
-    # DataFrame describing number of points in each interval that are flagged
-    resampled_flags_count = obs_flags.resample(
-        fx.interval_length, closed=closed, label=closed).sum()
+    # track number of points discarded before resampling in each interval
     to_discard_before_resample_count = to_discard_before_resample.resample(
         fx.interval_length, closed=closed, label=closed).sum()
 
     # Series to track if a given resampled interval should be discarded
-    to_discard = pd.Series(0, index=resampled_flags_count.index)
+    to_discard = pd.Series(False, index=to_discard_before_resample_count.index)
 
+    # will be used to determine threshold number of points
     interval_ratio = fx.interval_length / obs.interval_length
-    discard_after_resample_flags = [
-        f for f in quality_flags if not f.discard_before_resample
-    ]
-    for flag in discard_after_resample_flags:
-        threshold = flag.resample_threshold_percentage * interval_ratio
-        flagged = resampled_flags_count[flag.type] > threshold
-        to_discard += flagged
-        counts[flag.type] = flagged.sum()
+
+    for f in filter(lambda x: not x.discard_before_resample, quality_flags):
```

Comment:
This function is really hard to follow. Can it be split up into `_discard_before_resample` and `_discard_after_resample`? would probably need a final join on the counts and .loc on the series to drop the times from `_discard_before_resample`

Reply:
8a44022